libraries/chain/worker_proposal_evaluators.cpp

```diff
@@ -0,0 +1,63 @@
+#include <golos/protocol/worker_proposal_operations.hpp>
+#include <golos/protocol/exceptions.hpp>
+#include <golos/chain/steem_evaluator.hpp>
+#include <golos/chain/database.hpp>
+#include <golos/chain/steem_objects.hpp>
+#include <golos/chain/worker_proposal_objects.hpp>
+
+namespace golos { namespace chain {
+
+    void worker_proposal_create_evaluator::do_apply(const worker_proposal_create_operation& o) {
+        ASSERT_REQ_HF(STEEMIT_HARDFORK_0_20__1013, "worker_proposal_create_operation");
+
+        const auto& comment = _db.get_comment(o.author, o.permlink);
+
+        GOLOS_CHECK_LOGIC(comment.parent_author == STEEMIT_ROOT_POST_PARENT,
+            logic_exception::worker_proposal_can_be_created_only_on_post,
+            "Worker proposal can be created only on post");
+
+        const auto now = _db.head_block_time();
+
+        auto after_post_sec = (now - comment.created).to_seconds();
+        GOLOS_CHECK_LOGIC(after_post_sec >= GOLOS_WORKER_PROPOSAL_AFTER_POST_SEC,
+            logic_exception::too_early_creating_worker_proposal,
+            "Too early creating worker proposal for this post"); 
+
+        const auto& wpo_idx = db().get_index<worker_proposal_index, by_permlink>();
+        auto wpo_itr = wpo_idx.find(std::make_tuple(o.author, o.permlink));
+        if (wpo_itr != wpo_idx.end()) {
+            GOLOS_THROW_OBJECT_ALREADY_EXIST("worker_proposal_object", fc::mutable_variant_object()("author",o.author)("permlink",o.permlink));
```

Comment:
Why throw?
I can't chnage type of proposal?

Reply:
changing type leads to checks, is that change valid at current state. Is there any benefits from changing type? for me, it's completely different scenarios with no simple transformation from one to another